configuration/RealTimeTranslation/webResourceForCCaaS.js

```diff
@@ -0,0 +1,714 @@
+			var currentC2Language = C1WebResourceNamespace.dictForAllConversation[conversationId]?.['finalC2Lang'];
+			var languageUsedForUpdate = detectedLang;
+			if (detectedLangScore > 0.6 && currentC2Language != languageUsedForUpdate) {
+				if (this.enableLanguageDetectionWithHistoryMessages && highestScoredC2Language?.language) {
+					var highestScoredC2Language = highestScoredC2Language;
+					if (highestScoredC2Language?.language === languageUsedForUpdate ) {
+						consoleLogHelper(conversationId, "History message score based evaluation is enabled, the hightest scored language is matching detected language", {
+							dict: C1WebResourceNamespace.dictForAllConversation[conversationId],
+							highestScoredC2Language
+						});
+					}
+					else {
+						consoleLogHelper(conversationId, "History message score based evaluation is enabled, the hightest scored language is used to update", {
+							dict: C1WebResourceNamespace.dictForAllConversation[conversationId],
+							highestScoredC2Language,
+							currentC2Language
+						});
+						languageUsedForUpdate = highestScoredC2Language.language;
+					}
+				}
+				C1WebResourceNamespace.updateC2Language(conversationId, languageUsedForUpdate);
+			}
+		}
+		if (sourceLang == null) {
+			sourceLang = myJson[0]['detectedLanguage']["language"];
+		}
+		var result = {
+			translatedMessage: myJson[0]['translations'][0]['text'],
+			destinationLanguage: C1WebResourceNamespace.getOcLanguageCodeMapFromISO6391LanguageCode(String(myJson[0]['translations'][0]['to'])),
+			errorObject: errorObj,
+			sourceLanguage: C1WebResourceNamespace.getOcLanguageCodeMapFromISO6391LanguageCode(String(sourceLang))
+		};
+		consoleLogHelper(conversationId, "Returning translation result from Azure translator", {
+			result
+		});
+		return result;
+	},
+
+	////https://cloud.google.com/translate/docs/reference/rest/v2/translate
+	translateMessageInternalGoogle: async function (conversationId, message, messageSender, sourceLang, destLang) {
+		var errorObj = {
+			isError: false,
+			errorCode: null
+		};
+		consoleLogHelper(conversationId, "Trigger translateMessageInternalGoogle", {
+			message,
+			messageSender,
+			sourceLang,
+			destLang
+		})
+		//when we do not have even a single clue about C2's lang and c1 wants to send him message so destlang == null
+		if (destLang == null || sourceLang == destLang)
+			return {
+				translatedMessage: message,
+				destinationLanguage: C1WebResourceNamespace.getOcLanguageCodeMapFromISO6391LanguageCode(String(sourceLang)),
+				errorObject: errorObj,
+				sourceLanguage: C1WebResourceNamespace.getOcLanguageCodeMapFromISO6391LanguageCode(String(sourceLang))
+			};
+
+		var url = "https://translation.googleapis.com/language/translate/v2?format=text&key=" + C1WebResourceNamespace.googleTranslateApiClientSecret + "&target=" + destLang + "&q=" + message;
+		//translation happens here by calling api for translation engine

```

Comment:
can be written as:
`let url = `https://translation.googleapis.com/language/translate/v2?format=text&key=${C1WebResourceNamespace.googleTranslateApiClientSecret}&target=${destLang}&q=${encodeURIComponent(message)}`;

if (sourceLang) {
    url += `&source=${sourceLang}`;
}`